core/src/main/csharp/ch/cyberduck/core/taskdialog/TaskDialog.cs

```diff
@@ -309,19 +313,24 @@ unsafe static void CopyButtons(ref TASKDIALOG_BUTTON target, List<TASKDIALOG_BUT
                 handle = GCHandle.Alloc(this, GCHandleType.Normal);
                 copy.lpCallbackData = GCHandle.ToIntPtr(handle);
 
+                syncLockAcquired = syncEvent.WaitOne(250);
```

Comment:
Wait for 250ms before continuing regardless.
Whatever failed/hanged shouldn't block other threads/dialogs.
From my testing, this timeout hasn't been an issue.